language/move-vm/runtime/src/runtime.rs

```diff
@@ -106,11 +106,21 @@ impl VMRuntime {
         // load the script, perform verification
         let (main, type_params) = self.loader.load_script(&script, &ty_args, data_store)?;
 
-        // build the arguments list for the main and check the arguments are of restricted types
-        let first_param_opt = main.parameters().0.get(0);
-        if first_param_opt.map_or(false, |sig| is_signer_reference(sig)) {
-            args.insert(0, Value::transaction_argument_signer_reference(sender))
+        // Build the arguments list for the main and check the arguments are of restricted types.
+        // Signers are built up from left-to-right. Either all signer arguments are used, or no
+        // signer arguments can be be used by a script.
+        let parameters = &main.parameters().0;
+        if parameters.get(0).map_or(false, is_signer_reference) {
```

Comment:
nit. maybe bind this to a variable for readability. Just so its easy to see this is checking if a signer argument exists